### PR TITLE
Strip trailing periods from entries

### LINKED ISSUE #27: Validate: Strip trailing periods from entries
https://github.com/js-org/js.org/pull/8873#discussion_r1512115698

### PR DESCRIPTION
**Pull request by silaselisha**
Resolves #27

` const target = data.target.split('.');
        if(target[target.length - 1] === '') {
            target.splice(-1, 1);
        }`

### CODE REVIEW

File: src/util/cnames.js [hunk 1]
```diff
@@ -139,7 +139,13 @@ const generateCNAMEsFile = (cnames, file) => {
     for (const i in cnamesKeys) {
         const cname = cnamesKeys[i];
         const data = cnames[cname];
-        cnamesList.push(`  "${cname}": "${data.target}"${Number(i) === cnamesKeys.length - 1 ? '' : ','}${data.noCF ? ` ${data.noCF}` : ''}`)
+        
+        const target = data.target.split('.');
+        if(target[target.length - 1] === '') {
+            target.splice(-1, 1);
+        }
```
Comment (MattIPv4) [inline src/util/cnames.js:146]: Please do this where all the other modifications happen, in the parsing logic, not in the generation logic: https://github.com/js-org/js.org-cleanup/blob/25497f6457b0f7bd6b20d44631a3b3b054746ee9/src/util/cnames.js#L54-L93

Comment (silaselisha) [reply src/util/cnames.js:146]: noted.

**Review by silaselisha — COMMENTED** [review]
I was able to move the logic to the `parseCNAMEsFile`  and removed the same logic from the `generateCNAMEsFile`

File: src/util/cnames.js [hunk 2]
```diff
@@ -60,6 +60,13 @@ const parseCNAMEsFile = (content, context = {}) => {
         // Drop trailing slashes
         let target = match[2].replace(/\/+$/, '');
 
+        // Drop trailing periods
+        const targetArr = target.split('.');
+        if (targetArr[targetArr.length - 1] === '') {
+            targetArr.splice(-1, 1);
+            target = targetArr.join('.');
+        }
```
[the remark this answers is not in the document]
Comment (silaselisha) [reply src/util/cnames.js:68]: yes I'll handle this.

[the remark this answers is not in the document]
Comment (silaselisha) [reply src/util/cnames.js:68]: Thank you for the correction.

**Comment by silaselisha** [comment]
deleting the repository caused the deletion of this PR

**Pull request closed without merging**